src/SecurityManager.ts

```diff
@@ -400,19 +380,7 @@ async function doAccessSecretStorage(func: () => Promise<void>, forceReset: bool
                 getKeyBackupPassphrase: promptForBackupPassphrase,
             });
 
-            const keyId = Object.keys(secretStorageKeys)[0];
-            if (keyId && SettingsStore.getValue("feature_dehydration")) {
-                let dehydrationKeyInfo = {};
-                if (secretStorageKeyInfo[keyId] && secretStorageKeyInfo[keyId].passphrase) {
-                    dehydrationKeyInfo = { passphrase: secretStorageKeyInfo[keyId].passphrase };
-                }
-                logger.log("Setting dehydration key");
-                await cli.setDehydrationKey(secretStorageKeys[keyId], dehydrationKeyInfo, "Backup device");
-            } else if (!keyId) {
-                logger.warn("Not setting dehydration key: no SSSS key found");
-            } else {
-                logger.log("Not setting dehydration key: feature disabled");
-            }
+            await handleDeviceDehydration(cli);
```

Comment:
I refactored this method following the sonar report about `Cognitive Complexity`. I noticed that the part that handles device dehydration is only done when 4s is already setup. When it's not and when the bootstrap is done by `CreateSecretStorageDialog` it's not called. There might be an issue here, but as device dehydration has changed (now dehydration v2) might not be worth checking it?